Jenkinsfile

```diff
@@ -14,7 +14,7 @@ stage ("Builds") {
 			sh "scripts/run_unit_tests.sh"
 		}
 		stage ('Run integration tests') {
-			sh "sudo mount -t tmpfs tmpfs /tmp"
+			sh "sudo mount -t tmpfs -o size=80G tmpfs /tmp"
```

Comment:
@rbradford Without this, the CI is consistently failing.
What do you think about this? The problem is that it works because we know the VM has 130G of RAM. The default 63G allocated when executing `sudo mount -t tmpfs tmpfs /tmp` was not enough.
I'm a bit worried that we might even need more if we add some test consuming a bunch of disk, what do you think should be the best way to handle it?

Reply:
#453 might be an alternative (delete the files at the end of the tests.)

Reply:
@sboeuf I did a test on my VM (same config as the CI) with the patch from #453 and without this commit and the test passed fine.

I agree with your concerns about not using too much memory for the tmpfs.

Reply:
That being said I can see the usage jump to 60GiB of 63GiB available (right at the start of the test run when all the tests are able to complete in parallel.) One alternative could be to reduce our test parallelisation.

Reply:
I've bumped our CI agents up to the next size which have 192GiB of ram, giving a default tmpfs size of 96GiB which should be able to handle about 40 odd tests in parallel (we currently have 34 tests in total.)

I think an ideal solution would be to ratelimit the tests based on disk space. So we could catch the -ENOSPC and sleep that test for 60s and try again in a loop up to e.g. 5 times?

Reply:
Yes, the retry loop seems fine to me.
Maybe we could also come up with some simple math in the code, checking if the current test will have enough memory. Wait, this won't work because some other tests will run in parallel and we cannot be sure about the size being used at any point in time.

I think we should have a safety net about how much disk might be used at most. And what you mentioned about limiting the parallelization might actually be our best bet on controlling what's going on.

Reply:
So about this PR, I'm gonna remove the last commit, right? Since we use a bigger VM instance now.
BTW, bigger VM, means more RAM, but does it mean more vCPUs? Because if there's more vCPUs, then potentially more tests are going to run in parallel, and we might hit the tmpfs limit, right?

Reply:
Each test uses about 2GiB of disk so we still have some runway with 96GiB of tmpfs as we currently only have 34 tests.

Reply:
So yeh, drop the last commit for now.

Reply:
Ok! PR updated, should be ready to be merged now.